### ISSUE TITLE
Does corerun or coreclr work normally when it loads only dll file if there are .ni.dll and .dll?

### ISSUE THREAD

**Issue by YongseopKim** [dotnet/runtime#8866]
(This is assuming that `COMPlus_DumpJittedMethods=1` with checked built coreclr is checking whether ni file is loaded or not. This relates to https://github.com/dotnet/coreclr/issues/13772)

I don't know the policy of `corerun` or `coreclr`. So I'm asking. Does corerun or coreclr work normally when it loads only dll file if there are .ni.dll and .dll?

For example, (on 8/31 commits, I'm only testing on armel & legacy-arm32 and x64)
- clr: 050bfb50b14e066560004f6c65f4110a00efd64e
- fx: 0b16938f178f4cb509b7cdd0737037439b8b7d8c

There are each overlay & dll by using `crossgen` with `FNV`/`R2R` option. xxx-dll has `empty.dll` which just returns 1 on `Program::Main` method.
```
$ ls -1
fnv-dll
fnv-overlay
overlay
r2r-dll
r2r-overlay
```

If `corerun` or `coreclr` loaded ni.dll, `coreclr` doesn't do jitting `Program::Main`. However, coreclr does jitting on `FNV`/`R2R`.
```
$ grep NGE ./fnv-dll/empty.ni.dll             
Binary file ./fnv-dll/empty.ni.dll matches

$ COMPlus_DumpJittedMethods=1 ./fnv-overlay/corerun ./fnv-dll/empty.dll
Compiling    0 Program::Main, IL size = 7, hsh=0xabb389cf
$ echo $?
1

$ grep RTR ./r2r-dll/empty.ni.dll             
Binary file ./r2r-dll/empty.ni.dll matches

$ COMPlus_DumpJittedMethods=1 ./r2r-overlay/corerun ./r2r-dll/empty.dll
Compiling    0 Program::Main, IL size = 7, hsh=0xabb389cf
$ echo $?
1
```

However, these results are different from older coreoverlay(on march~april).
- clr: 958279de27e7f3c30ad53e35277653d1ec0f1e6e
- fx: d19c909e93917386cfdcdfb79e9492ab9c756abb

```
$ grep NGE ./fnv-dll/empty.ni.dll 
Binary file ./fnv-dll/empty.ni.dll matches

$ COMPlus_DumpJittedMethods=1 ./fnv-overlay/corerun ./fnv-dll/empty.dll
# nothing prints. it means coreclr jits nothing.
$ echo $?
1

$ grep RTR ./r2r-dll/empty.ni.exe 
Binary file ./r2r-dll/empty.ni.exe matches

$ COMPlus_DumpJittedMethods=1 ./r2r-overlay/corerun ./r2r-dll/empty.dll
# nothing prints. it means coreclr jits nothing.
$ echo $?
1
```
It seems that the former acts weird and the latter acts normally.

In short, is it normal that only dll file is loaded if there are also ni file?

**Comment by jkotas** [comment]
Fragile ngen support has been deleted.
